### PR TITLE
SONARPY-1662 Fix FP on S117 when a class is assigned to a variable

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**

### CODE REVIEW

File: python-checks/src/test/resources/checks/itemOperationsTypeCheck/itemOperations_delitem.py [hunk 1]
```diff
@@ -180,3 +180,12 @@ class ExtendedMock(MagicMock):
...
     def custom_mock():
         del ExtendedMock()[42]
+
+
+class SomeClass:
+    ...
+
+def get_state_fp():
+    some_class = SomeClass()
+    state = SomeClass.__getstate__(some_class)  # __getstate__ usually returns a dict but is annotated to return 'object' in Typeshed
```
Comment (guillaume-dequenne-sonarsource) [inline python-checks/src/test/resources/checks/itemOperationsTypeCheck/itemOperations_delitem.py:190]: Unfortunately, I believe we'd need to update the annotation in Typeshed to fix this. I think we can accept the FP for now.

File: its/ruling/src/test/resources/expected/python-S5644.json [hunk 2]
```diff
@@ -1,49 +1,53 @@
 {
```
Comment (guillaume-dequenne-sonarsource) [inline its/ruling/src/test/resources/expected/python-S5644.json:1]: Unfortunately changes here are FPs (documented in the tests for S5644) due to a wrong stub annotation for `type.__getstate__`

File: its/ruling/src/test/resources/expected/python-S3699.json [hunk 3]
```diff
@@ -1,19 +1,22 @@
 {
```
Comment (guillaume-dequenne-sonarsource) [inline its/ruling/src/test/resources/expected/python-S3699.json:1]: I believe the change here is a TP.

**Review by ghislainpiot — APPROVED** [review]
Looks good to me!

**Pull request merged**